### PR TITLE
Typo and style

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Few fixes staging on my repo

##### Checklist
- [X] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/SSL_read_early_data.pod [hunk 1]
```diff
@@ -181,7 +181,7 @@ with then the lower of the two values will apply.
 The whole purpose of early data is to enable a client to start sending data to
 the server before a full round trip of network traffic has occurred. Application
 developers should ensure they consider optimisation of the underlying TCP socket
-to obtain a performant solution. For example Nagle's algorithm is commonly used
+to obtain a efficient solution. For example Nagle's algorithm is commonly used
```
Comment (levitte) [inline doc/man3/SSL_read_early_data.pod]: a*n* efficient

File: doc/man3/SSL_read_early_data.pod [hunk 2]
```diff
@@ -181,7 +181,7 @@ with then the lower of the two values will apply.
 The whole purpose of early data is to enable a client to start sending data to
 the server before a full round trip of network traffic has occurred. Application
 developers should ensure they consider optimisation of the underlying TCP socket
-to obtain a performant solution. For example Nagle's algorithm is commonly used
+to obtain an efficient solution. For example Nagle's algorithm is commonly used
```
Comment (kaduk) [inline doc/man3/SSL_read_early_data.pod]: This one seems like a matter of taste, but I don't really prefer one over the other, myself.

File: include/openssl/ssl.h [hunk 3]
```diff
@@ -217,11 +217,9 @@ typedef struct srtp_protection_profile_st {
 
 DEFINE_STACK_OF(SRTP_PROTECTION_PROFILE)
 
-typedef int (*tls_session_ticket_ext_cb_fn) (SSL *s,
-                                             const unsigned char *data,
+typedef int (*tls_session_ticket_ext_cb_fn) (SSL *s, const unsigned char *data,
                                              int len, void *arg);
```
Comment (kaduk) [inline include/openssl/ssl.h]: Huh, the old form is surprising.  I could understand breaking early to get 'data' and 'len' on the same line, but the old form seems to have not had any real benefit.

File: include/openssl/ssl.h [hunk 4]
```diff
@@ -541,7 +539,7 @@ typedef int (*SSL_verify_cb)(int preverify_ok, X509_STORE_CTX *x509_ctx);
  */
 
 unsigned long SSL_CTX_get_options(const SSL_CTX *ctx);
-unsigned long SSL_get_options(const SSL* s);
+unsigned long SSL_get_options(const SSL *ssl);
```
Comment (kaduk) [inline include/openssl/ssl.h]: Please revert this chunk; ```SSL *s``` is the prevailing style here.

File: include/openssl/ssl.h [hunk 5]
```diff
@@ -685,11 +682,11 @@ void SSL_CTX_sess_set_get_cb(SSL_CTX *ctx,
                                                              *data, int len,
                                                              int *copy));
 SSL_SESSION *(*SSL_CTX_sess_get_get_cb(SSL_CTX *ctx)) (struct ssl_st *ssl,
-                                                       const unsigned char *data,
-                                                       int len, int *copy);
+                                                       const unsigned char
+                                                       *data, int len,
+                                                       int *copy);
```
Comment (kaduk) [inline include/openssl/ssl.h]: For these sort of callback getters, I'd almost consider a more radical solution that indents the getter arguments only a few columns more than, say, the start of the getter function name.  But definitely wait for a second opinion before implementing that!

File: include/openssl/ssl.h [hunk 6]
```diff
@@ -740,9 +737,9 @@ void SSL_get0_next_proto_negotiated(const SSL *s, const unsigned char **data,
 # endif
 
 __owur int SSL_select_next_proto(unsigned char **out, unsigned char *outlen,
-                          const unsigned char *in, unsigned int inlen,
-                          const unsigned char *client,
-                          unsigned int client_len);
+                                 const unsigned char *in, unsigned int inlen,
+                                 const unsigned char *client,
+                                 unsigned int client_len);
```
Comment (kaduk) [inline include/openssl/ssl.h]: Looks like the ```__owur ``` addition was a little sloppy.

File: include/openssl/ssl.h [hunk 7]
```diff
@@ -1485,11 +1483,12 @@ void SSL_set_verify_depth(SSL *s, int depth);
 void SSL_set_cert_cb(SSL *s, int (*cb) (SSL *ssl, void *arg), void *arg);
 # ifndef OPENSSL_NO_RSA
 __owur int SSL_use_RSAPrivateKey(SSL *ssl, RSA *rsa);
-__owur int SSL_use_RSAPrivateKey_ASN1(SSL *ssl, const unsigned char *d, long len);
+__owur int SSL_use_RSAPrivateKey_ASN1(SSL *ssl, const unsigned char *d,
+                                      long len);
```
Comment (kaduk) [inline include/openssl/ssl.h]: My count has the final semicolon in column 79; I thought that was okay in our style as-is.

File: include/openssl/ssl.h [hunk 8]
```diff
@@ -1585,12 +1588,13 @@ int SSL_SESSION_up_ref(SSL_SESSION *ses);
 void SSL_SESSION_free(SSL_SESSION *ses);
 __owur int i2d_SSL_SESSION(SSL_SESSION *in, unsigned char **pp);
 __owur int SSL_set_session(SSL *to, SSL_SESSION *session);
```
Comment (kaduk) [inline include/openssl/ssl.h]: Why leave this one as ```SSL *to```?

File: include/openssl/ssl.h [hunk 9]
```diff
@@ -1585,12 +1588,13 @@ int SSL_SESSION_up_ref(SSL_SESSION *ses);
...
-int SSL_CTX_remove_session(SSL_CTX *, SSL_SESSION *c);
-__owur int SSL_CTX_set_generate_session_id(SSL_CTX *, GEN_SESSION_CB);
-__owur int SSL_set_generate_session_id(SSL *, GEN_SESSION_CB);
-__owur int SSL_has_matching_session_id(const SSL *ssl, const unsigned char *id,
-                                unsigned int id_len);
+int SSL_CTX_add_session(SSL_CTX *ctx, SSL_SESSION *session);
+int SSL_CTX_remove_session(SSL_CTX *ctx, SSL_SESSION *session);
+__owur int SSL_CTX_set_generate_session_id(SSL_CTX *ctx, GEN_SESSION_CB cb);
+__owur int SSL_set_generate_session_id(SSL *ssl, GEN_SESSION_CB cb);
+__owur int SSL_has_matching_session_id(const SSL *ssl,
```
Comment (kaduk) [inline include/openssl/ssl.h]: I think we still prefer ```SSL *s``` here.

File: include/openssl/ssl.h [hunk 10]
```diff
@@ -1612,15 +1616,15 @@ void SSL_CTX_set_cert_cb(SSL_CTX *c, int (*cb) (SSL *ssl, void *arg),
                          void *arg);
 # ifndef OPENSSL_NO_RSA
 __owur int SSL_CTX_use_RSAPrivateKey(SSL_CTX *ctx, RSA *rsa);
-__owur int SSL_CTX_use_RSAPrivateKey_ASN1(SSL_CTX *ctx, const unsigned char *d,
```
Comment (kaduk) [inline include/openssl/ssl.h]: This comma is in column 79 (one-indexed); I don't think it needs to change.

File: include/openssl/ssl.h [hunk 11]
```diff
@@ -1634,19 +1638,21 @@ void *SSL_get_default_passwd_cb_userdata(SSL *s);
...
-__owur int SSL_CTX_set_session_id_context(SSL_CTX *ctx, const unsigned char *sid_ctx,
-                                   unsigned int sid_ctx_len);
+__owur int SSL_CTX_set_session_id_context(SSL_CTX *ctx,
+                                          const unsigned char *sid_ctx,
+                                          unsigned int sid_ctx_len);
 
 SSL *SSL_new(SSL_CTX *ctx);
 int SSL_up_ref(SSL *s);
 int SSL_is_dtls(const SSL *s);
-__owur int SSL_set_session_id_context(SSL *ssl, const unsigned char *sid_ctx,
```
Comment (kaduk) [inline include/openssl/ssl.h]: And this is in column 77(!)

File: include/openssl/ssl.h [hunk 12]
```diff
@@ -1634,19 +1638,21 @@ void *SSL_get_default_passwd_cb_userdata(SSL *s);
...
+                                      unsigned int sid_ctx_len);
 
-__owur int SSL_CTX_set_purpose(SSL_CTX *s, int purpose);
-__owur int SSL_set_purpose(SSL *s, int purpose);
-__owur int SSL_CTX_set_trust(SSL_CTX *s, int trust);
-__owur int SSL_set_trust(SSL *s, int trust);
+__owur int SSL_CTX_set_purpose(SSL_CTX *ctx, int purpose);
+__owur int SSL_set_purpose(SSL *ssl, int purpose);
+__owur int SSL_CTX_set_trust(SSL_CTX *ctx, int trust);
+__owur int SSL_set_trust(SSL *ssl, int trust);
```
Comment (kaduk) [inline include/openssl/ssl.h]: ```SSL *s``` again

File: include/openssl/ssl.h [hunk 13]
```diff
@@ -1737,8 +1743,8 @@ void SSL_free(SSL *ssl);
 __owur int SSL_waiting_for_async(SSL *s);
 __owur int SSL_get_all_async_fds(SSL *s, OSSL_ASYNC_FD *fds, size_t *numfds);
 __owur int SSL_get_changed_async_fds(SSL *s, OSSL_ASYNC_FD *addfd,
-                                     size_t *numaddfds, OSSL_ASYNC_FD *delfd,
-                                     size_t *numdelfds);
+                                     size_t *numaddfds,
+                                     OSSL_ASYNC_FD *delfd, size_t *numdelfds);
```
Comment (kaduk) [inline include/openssl/ssl.h]: Going from column 77 (old) to column 78 (new)?  What's the motivation for this change?

File: include/openssl/ssl.h [hunk 14]
```diff
@@ -2008,13 +2015,14 @@ int SSL_bytes_to_cipher_list(SSL *s, const unsigned char *bytes, size_t len,
...
+__owur int SSL_set_session_ticket_ext_cb(SSL *s,
+                                         tls_session_ticket_ext_cb_fn cb,
+                                         void *arg);
 
 /* Pre-shared secret session resumption functions */
 __owur int SSL_set_session_secret_cb(SSL *s,
-                              tls_session_secret_cb_fn tls_session_secret_cb,
-                              void *arg);
+                                     tls_session_secret_cb_fn
+                                     tls_session_secret_cb, void *arg);
```
Comment (kaduk) [inline include/openssl/ssl.h]: I would prefer to avoid splitting the type and name across lines like this.
Similar comments as to callback getters (above) apply, and I suppose we also retain the option of not specifying a name at all.

File: include/openssl/ssl.h [hunk 15]
```diff
@@ -2023,8 +2031,7 @@ void SSL_CTX_set_not_resumable_session_callback(SSL_CTX *ctx,
 
 void SSL_set_not_resumable_session_callback(SSL *ssl,
                                             int (*cb) (SSL *ssl,
-                                                       int
-                                                       is_forward_secure));
+                                                       int is_forward_secure));
```
Comment (kaduk) [inline include/openssl/ssl.h]: Is this going from column 79 to column 83?!

File: include/openssl/ssl.h [hunk 16]
```diff
@@ -2082,8 +2090,8 @@ int DTLSv1_listen(SSL *s, BIO_ADDR *client);
  * May return a negative integer if an error occurs.
  * A connection should be aborted if the SCTs are deemed insufficient.
  */
-typedef int(*ssl_ct_validation_cb)(const CT_POLICY_EVAL_CTX *ctx,
-                                   const STACK_OF(SCT) *scts, void *arg);
+typedef int (*ssl_ct_validation_cb) (const CT_POLICY_EVAL_CTX * ctx,
+                                     const STACK_OF(SCT) *scts, void *arg);
```
Comment (kaduk) [inline include/openssl/ssl.h]: It looks like we prefer to keep the ```)(``` nestled together for the function (type) arguments.
Also, no space in ```*ctx```.

File: include/openssl/ssl.h [hunk 17]
```diff
@@ -2170,7 +2178,7 @@ void SSL_CTX_set0_ctlog_store(SSL_CTX *ctx, CTLOG_STORE *logs);
  */
 const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
 
-# endif /* OPENSSL_NO_CT */
+# endif                         /* OPENSSL_NO_CT */
```
Comment (kaduk) [inline include/openssl/ssl.h]: Huh, this is the only instance in ssl.h of endif-with-comment.
I guess your change is consistent with ssl_locl.h, so I don't really have grounds to object.

File: include/openssl/ssl.h [hunk 18]
```diff
@@ -2233,21 +2241,21 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
...
 void SSL_set_security_callback(SSL *s,
-                               int (*cb) (const SSL *s, const SSL_CTX *ctx, int op,
-                                          int bits, int nid, void *other,
-                                          void *ex));
-int (*SSL_get_security_callback(const SSL *s)) (const SSL *s, const SSL_CTX *ctx, int op,
-                                                int bits, int nid,
+                               int (*cb) (const SSL *s, const SSL_CTX *ctx,
+                                          int op, int bits, int nid,
+                                          void *other, void *ex));
+int (*SSL_get_security_callback(const SSL *s)) (const SSL *s, const SSL_CTX *ctx,
```
Comment (kaduk) [inline include/openssl/ssl.h]: Isn't this trailing comma still column 81?

File: include/openssl/ssl.h (hunk 4 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: OK, done.
I think (and wish) that one letter should be forbidden for non-POD type.
It 's a pain to track them in the code .
May be a future addition to Style Guideline ? ;)

File: include/openssl/ssl.h (hunk 5 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: A better solution `struct ssl_st* ->   SSL*` ...
plus a typedef on callback functions  ... 

I did not dig the origins of this kind of declarations ...

File: include/openssl/ssl.h (hunk 7 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: `;` is at col 82 in my editor

File: include/openssl/ssl.h (hunk 8 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Less annoying IMHO, it set a `session` into `to` ... clear for me ;)
Feel free to rename it,

File: include/openssl/ssl.h (hunk 10 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: ok, fixed

File: include/openssl/ssl.h (hunk 12 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: One letter naming  :(

File: include/openssl/ssl.h (hunk 13 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: indent tool choice may be, I don't remenber

File: include/openssl/ssl.h (hunk 15 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Not on my side.

File: include/openssl/ssl.h (hunk 18 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: yes it is ...

File: doc/man3/SSL_read_early_data.pod (hunk 2 above)
Comment (FdaSilvaYY) [reply doc/man3/SSL_read_early_data.pod]: 'performant' is missing in a some dictionnaries

**Comment by FdaSilvaYY** [comment]
Should I split out `ssl.h` corrections into his standalone PR to ease the Approval ?

**Comment by FdaSilvaYY** [comment]
Rebased to latest master

**Comment by FdaSilvaYY** [comment]
Comments fixed, and PR rebased to last master.

File: ssl/record/rec_layer_s3.c [hunk 19]
```diff
@@ -920,12 +920,12 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
...
-                   /*
-                    * We also need next the amount of bytes written to this
-                    * sub-packet
-                    */
+        if (!WPACKET_reserve_bytes(thispkt, SSL_RT_MAX_CIPHER_BLOCK_SIZE,
+                                   NULL)
+               /*
+                * We also need next the amount of bytes written to this
+                * sub-packet
+                */
```
Comment (levitte) [inline ssl/record/rec_layer_s3.c]: nit pick.  The comment needs to move one space to the right

Comment (FdaSilvaYY) [reply ssl/record/rec_layer_s3.c]: fixed

**Comment by kaduk** [comment]
reconfirm +1
I'll merge.

**Comment by kaduk** [comment]
Merged to master; closing
